### ISSUE TITLE
no-unused-variable false positive when used only in string literal

### ISSUE THREAD

**Issue by timkinnane** [palantir/tslint#3726]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: 2.7.1
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
  constructor (private host = 'localhost:3000') {
    super()
    this.asteroid = new Asteroid({
      endpoint: `ws://${host}/websocket`,
      SocketConstructor: ws
    })
    this.asteroid.on('connected', () => this.emit('connected'))
    this.asteroid.on('reconnected', () => this.emit('reconnected'))
  }
```

#### Actual behavior

ERROR: Property 'host' is declared but its value is never read.

#### Expected behavior

Should pass lint.
